src/pcre2test.c

```diff
@@ -9250,20 +9260,20 @@ for (gmatched = 0;; gmatched++)
 
   /* The result of the match is now in capcount. First handle a successful
   match. If pp was forced to be NULL (to test NULL handling) it will have been
-  treated as an empty string if the length was zero. So re-create that for
-  outputting. Don't just point to "" because that leads to a "loss of const"
-  warning. */
+  treated as an empty string if the length was zero. So, re-create that for
+  outputting, preserving the invariant that pp is a valid pointer to a region
+  of length len followed by a null. */
 
   if (capcount >= 0)
     {
     if (pp == NULL)
       {
 #ifdef SUPPORT_VALGRIND
       /* Mark the start of dbuffer addressable again. */
-      VALGRIND_MAKE_MEM_UNDEFINED(dbuffer, 1);
+      VALGRIND_MAKE_MEM_UNDEFINED(dbuffer, code_unit_size);
 #endif
       pp = dbuffer;
-      pp[0] = 0;
+      memset(pp, 0, code_unit_size);
```

Comment:
Another lucky bug. We are writing only a single byte here, not a full code unit (on 16-bit and 32-bit libraries). However, the whole block of code is just belt-and-braces stuff. Even if you remove it totally (or write `0xDEADBEEF` instead of null) the tests all pass absolutely fine, even under valgrind.

Basically we don't need the trailing null, so it's harmless that we didn't write it correctly.